setup.py

```diff
@@ -1,12 +1,39 @@
 #:!/usr/bin/env python
 from setuptools import find_packages, setup
+import subprocess
+import re
 
 VERSION = "0.0.1"  # was previously None
 
 # add the README.md file to the long_description
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
+def get_cuda_version():
+    try:
+        output = subprocess.check_output(["nvcc", "--version"]).decode()
+        match = re.search(r"release (\d+\.\d+)", output)
+        if match:
+            return float(match.group(1))
+        else:
+            return None
+    except FileNotFoundError:
+        raise FileNotFoundError("nvcc not found. Install CUDA and make sure it's in your path.")
+        return None
+
+cuda_version = get_cuda_version()
+
+if cuda_version is not None:
+    if 11.0 <= cuda_version < 12.0:
+        dali = "nvidia-dali-cuda110"
+    elif 12.0 <= cuda_version < 13.0:
+        dali = "nvidia-dali-cuda120"
```

Comment:
should we fix this to be `"nvidia-dali-cuda120==1.25.0"` to match our pinned version from before?

Reply:
for 120 it's fine. cuda110 has a different release; 1.16.0 I think, in my tests. We never had issues with DALI versions before so I don't worry too much about it... If you have strong preferences I can pin each separately